compiler/il/VectorOperations.enum

```diff
@@ -739,6 +739,553 @@ VECTOR_OPERATION_MACRO(\
+   /* .ifCompareOpCode         = */ TR::BadILOp, \
+   /* .description             =    vector masked logical OR */ \
+)
+VECTOR_OPERATION_MACRO(\
+   /* .operation               = */ vmorUnchecked, \
+   /* .name                    = */ "vmorUnchecked", \
+   /* .properties1             = */ ILProp1::VectorMasked, \
+   /* .properties2             = */ ILProp2::ValueNumberShare | ILProp2::SupportedForPRE, \
+   /* .properties3             = */ 0, \
+   /* .properties4             = */ 0, \
+   /* .dataType                = */ TR::NoType, \
+   /* .typeProperties          = */ ILTypeProp::VectorResult, \
+   /* .childProperties         = */ THREE_SAME_CHILD(ILChildProp::UnspecifiedChildType), \
+   /* .swapChildrenOperation   = */ TR::vBadOperation, \
+   /* .reverseBranchOperation  = */ TR::vBadOperation, \
+   /* .booleanCompareOpCode    = */ TR::BadILOp, \
+   /* .ifCompareOpCode         = */ TR::BadILOp, \
+   /* .description             =    the same as vmor but is valid for FP types as well*/ \
+)
+VECTOR_OPERATION_MACRO(\
+   /* .operation               = */ vmreductionAdd, \
+   /* .name                    = */ "vmreductionAdd", \
+   /* .properties1             = */ ILProp1::Commutative | ILProp1::Associative | ILProp1::VectorMasked, \
+   /* .properties2             = */ ILProp2::ValueNumberShare | ILProp2::SupportedForPRE, \
+   /* .properties3             = */ 0, \
+   /* .properties4             = */ 0, \
+   /* .dataType                = */ TR::NoType, \
+   /* .typeProperties          = */ TR::NoType, \
+   /* .childProperties         = */ TWO_SAME_CHILD(ILChildProp::UnspecifiedChildType), \
+   /* .swapChildrenOperation   = */ TR::vBadOperation, \
+   /* .reverseBranchOperation  = */ TR::vBadOperation, \
+   /* .booleanCompareOpCode    = */ TR::BadILOp, \
+   /* .ifCompareOpCode         = */ TR::BadILOp, \
+   /* .description             =    vector masked add reduction */ \
+)
+VECTOR_OPERATION_MACRO(\
+   /* .operation               = */ vmreductionAnd, \
+   /* .name                    = */ "vmreductionAnd", \
+   /* .properties1             = */ ILProp1::Commutative | ILProp1::Associative | ILProp1::VectorMasked, \
+   /* .properties2             = */ ILProp2::ValueNumberShare | ILProp2::SupportedForPRE, \
+   /* .properties3             = */ 0, \
+   /* .properties4             = */ 0, \
+   /* .dataType                = */ TR::NoType, \
+   /* .typeProperties          = */ TR::NoType, \
+   /* .childProperties         = */ TWO_SAME_CHILD(ILChildProp::UnspecifiedChildType), \
+   /* .swapChildrenOperation   = */ TR::vBadOperation, \
+   /* .reverseBranchOperation  = */ TR::vBadOperation, \
+   /* .booleanCompareOpCode    = */ TR::BadILOp, \
+   /* .ifCompareOpCode         = */ TR::BadILOp, \
+   /* .description             =    vector masked and reduction */ \
+)
+VECTOR_OPERATION_MACRO(\
+   /* .operation               = */ vmreductionFirstNonZero, \
+   /* .name                    = */ "vmreductionFirstNonZero", \
```

Comment:
FirstNonZero is a reduction operation by definition, so there shouldn't be two opcodes for this.

Reply:
Interestingly, there is a lanewise FirstNonZero. I think it takes the lane value of the first vector it is't non-zero and the value of the second vector if it is.